Mesh_3/include/CGAL/Mesh_3/generate_label_weights.h

```diff
@@ -137,7 +137,7 @@ void convert_itk_to_image_3(itk::Image<Image_word_type, 3>* const itk_img,
             itk_img->GetBufferPointer() + size,
             img_ptr);
 
-  if(filename != "")
+  if(std::strlen(filename)!=0)
```

Comment:
[`std::strlen` is unsafe](https://stackoverflow.com/q/5935413/1728537), and `filename` is a user-provided string. I suggest using `std::string` instead:
```suggestion
  if(!std::string(filename).empty())
```

Actually, I do not really understand why `filename` has default empty value, and why there is a test for `filename` before the call to `_writeImage`. Maybe the right patch is removing that line (and fix the indentation):
```suggestion
```